tests/tekton-resources/tasks/setup/karpenter/helm-karpenter-install.yaml

```diff
@@ -0,0 +1,136 @@
+    description: EKS cluster endpoint URL for kubectl configuration
+  workspaces:
+  - name: config
+  steps:
+  - name: install-karpenter
+    image: alpine/k8s:1.30.2
+    script: |
+      echo "Starting Karpenter Installation Task"
+      echo "===================================="
+
+      # Authenticate with ECR for Helm registry access
+      echo "[INFO] Authenticating with ECR registry..."
+      aws ecr get-login-password --region us-west-2 | helm registry login --username AWS --password-stdin $(params.karpenter-ecr-repo)
+      echo "[SUCCESS] Successfully authenticated with ECR"
+      echo ""
+
+      # Configure kubectl to connect to the EKS cluster
+      echo "[INFO] Configuring kubectl for cluster access..."
+      aws eks update-kubeconfig --name $(params.cluster-name) --endpoint $(params.endpoint)
+      echo "[SUCCESS] Successfully configured kubectl"
+      echo ""
+
+      # Verify karpenter-system nodegroup exists
+      echo "[INFO] Verifying karpenter-system nodegroup..."
+      aws eks describe-nodegroup --cluster-name $(params.cluster-name) --endpoint $(params.endpoint) --nodegroup-name karpenter-system-large
+      echo ""
+      
+      # Capture cluster state before installation for troubleshooting
+      echo "[INFO] Capturing cluster state before Karpenter installation..."
+      echo "----------------------------------------"
+      
+      echo "[INFO] Current cluster nodes:"
+      kubectl get nodes -o wide
+      echo ""
+      
+      echo "[INFO] Current pods across all namespaces:"
+      kubectl get pods -A -o wide
+      echo ""
+
+      echo "[INFO] Current deployments across all namespaces:"
+      kubectl get deployments -A -o wide
+      echo "----------------------------------------"
+      echo ""
+      
+      # Install Karpenter using Helm with optimized configuration
+      echo "[INFO] Installing Karpenter with Helm..."
+      echo ""
+      
+      helm upgrade --install karpenter oci://$(params.karpenter-ecr-repo)/karpenter/karpenter --version $(params.karpenter-version) \
+        --namespace "karpenter" \
+        --create-namespace \
+        --set "settings.clusterName=$(params.cluster-name)" \
+        --set "settings.interruptionQueue=" \
+        --set "settings.eksControlPlane=true" \
+        --set-string "settings.awsCreateQPS=60" \
+        --set "settings.featureGates.disableMetricsControllers=true" \
+        --set "settings.featureGates.nodeRepair=true" \
+        --set settings.featureGates.reservedCapacity="true" \
+        --set settings.featureGates.spotToSpotConsolidation="true" \
+        --set settings.featureGates.disableMetricsControllers="true" \
+        --set settings.preferencePolicy=Ignore \
+        --set "serviceAccount.annotations.eks\.amazonaws\.com/role-arn=arn:aws:iam::$(params.aws-account-id):role/KarpenterControllerRole-$(params.cluster-name)" \
+        --set controller.resources.requests.cpu=60 \
+        --set controller.resources.requests.memory=200Gi \
```

Comment:
resource allocation to controller, should be dependent on the size of the cluster ? can we parameterize these ?